### PR TITLE
[Form] Remove deprecated implementation of ChoiceLoaderInterface in intl forms

### PR DESCRIPTION
**Pull request by yceruto**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

See previous PR https://github.com/symfony/symfony/pull/26825 (4.1)

### CODE REVIEW

**Comment by yceruto** [comment]
Status: Needs Work

**Comment by yceruto** [comment]
(Tests failures unrelated)

Status: Needs Review

File: src/Symfony/Component/Form/Extension/Core/Type/CountryType.php [hunk 1]
```diff
@@ -12,38 +12,23 @@
...
-    private $choiceList;
-
     /**
      * {@inheritdoc}
      */
     public function configureOptions(OptionsResolver $resolver)
     {
         $resolver->setDefaults([
-            'choice_loader' => function (Options $options) {
+            'choice_loader' => static function (Options $options) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Extension/Core/Type/CountryType.php]: these are for 4.4

**Review by nicolas-grekas — CHANGES_REQUESTED** [review]
please move the added `static` to a PR for 4.4

File: src/Symfony/Component/Form/Extension/Core/Type/CountryType.php (hunk 1 above)
Comment (yceruto) [reply src/Symfony/Component/Form/Extension/Core/Type/CountryType.php]: Remove from this PR, later I will do the changes for 4.4

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @yceruto.

**Pull request merged**